packages/world/src/modules/keysintable/KeysInTableDynamicFieldIndex.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { StoreSwitch } from "@latticexyz/store/src/StoreSwitch.sol";
+import { Bytes } from "@latticexyz/store/src/Bytes.sol";
+
+import { KeysInTable, KeysInTableTableId } from "./tables/KeysInTable.sol";
+
+/**
+ * Based on `KeysInTable`, but allows dynamically passing fieldIndex.
+ * Dynamic fieldIndex relies on all fields having the same schema type.
+ */
+library KeysInTableDynamicFieldIndex {
+  function pushKeys(uint8 fieldIndex, bytes32 sourceTable, bytes32 _element) internal {
```

Comment:
nit - why keys, not key?
```suggestion
  function pushKey(uint8 fieldIndex, bytes32 sourceTable, bytes32 _element) internal {
```

Reply:
tbf it's neither key nor keys, but keyParts - renamed it
it's keyParts (not keyPart) because the fields are arrays
it's keyParts (not keys) because the fields are each a single element of a keyTuple
pushKeys is slightly misleading and maybe I should make it `push* -> pushTo*` and `pop* -> popFrom*` (I didn't because it's more verbose)

Reply:
oh also this table is mostly copypasted from `KeysInTable` and I don't think they should diverge more than necessary, so I changed `mud.config.ts` too